.github/actions/setup-dependencies/action.yml

```diff
@@ -13,41 +13,46 @@ inputs:
 runs:
   using: "composite"
   steps:
-    # ———— Linux and macOS ———— #
-    - name: Set Linux MAKEFLAGS
-      run: echo "MAKEFLAGS=-j$(nproc) V=1" >> $GITHUB_ENV
-      shell: bash
-      if: runner.os == 'Linux'
-
-    - name: Set macOS MAKEFLAGS
-      run: echo "MAKEFLAGS=-j$(sysctl -n hw.ncpu) V=1" >> $GITHUB_ENV
-      shell: bash
-      if: runner.os == 'macOS'
-
-    - name: Install brew dependencies
-      run: brew install node php lmdb mcpp || true
-      shell: bash
-      if: runner.os == 'macOS'
-
-    # - name: Add Ruby and Python interpreters from brew to PATH
-    #   run: |
-    #     echo "$(brew --prefix ruby)/bin" >> $GITHUB_PATH
-    #     echo "$(brew --prefix python3)/bin" >> $GITHUB_PATH
-    #   shell: bash
-    #   if: runner.os == 'macOS'
-
+    #
+    # General Setup (Python, Ruby, Node, Java, .NET, etc.)
+    #
     - uses: actions/setup-python@v5
       with:
-        python-version: "3.12"
-      if: runner.os == 'macOS'
+        python-version: "3.13"
 
     - uses: ruby/setup-ruby@v1
       with:
         ruby-version: "3.3"
-      if: runner.os == 'macOS'
+
+    - uses: actions/setup-node@v4
+      with:
+        node-version: "20"
+
+    - name: Setup Oracle Java 17
+      uses: actions/setup-java@v4
+      with:
+        distribution: "oracle"
+        java-version: "17"
+
+    - name: Setup .NET 8
+      uses: actions/setup-dotnet@v4
+      with:
+        dotnet-version: 8.x
 
     - name: Install testing dependencies from pip
-      run: python3 -m pip install --break-system-packages --user passlib cryptography numpy
+      run: python3 -m pip install passlib cryptography numpy
```

Comment:
Isn't `--break-system-packages` required with this version?

Reply:
It shouldn't be since we're now using a python installed form the action.